smart_open/smart_open_lib.py

```diff
@@ -355,6 +356,9 @@ class ParseUri(object):
       * file:///home/user/file
       * file:///home/user/file.bz2
 
+    NOTE: hdfs://path/file does no longer work as it is against the URI
```

Comment:
Can you add more information to comment (when this happens, what HDFS version affected, etc)?

Reply:
This is a generic change that affects all HDFS versions. They do support hdfs://host/path/file URI format. Using hdfs://path/file in smart_open violates this and also violates the URI specification (RFC3986) where the hostname part is always after the first two / characters. The correct way to refer to local content is hdfs:///path/file instead of hdfs://path/file.